.github/workflows/go-work-sync.yaml

```diff
@@ -20,33 +20,25 @@ jobs:
     # We only run this for pull requests from the same repository.  This is
     # important for security reasons, as we use pull_request_target.
     if: github.event.pull_request.head.repo.full_name == github.repository
+    permissions:
+      contents: read
+      id-token: write # Required for ./.github/actions/get-token
     runs-on: ubuntu-latest
     steps:
-    # Because the GitHub-provided token doesn't trigger further actions runs,
-    # try to use a secret if available.
-    - name: Determine checkout token
-      id: has-token
-      run: echo "has-token=$HAS_TOKEN" >> "$GITHUB_OUTPUT"
-      env:
-        # Temporarily disable use of token; we don't have a correct token set up
-        # (so we fail to push), so using the GitHub-provided token that doesn't
-        # trigger subsequent checks is better than failing to push.
-        HAS_TOKEN: ${{ false && secrets.RUN_WORKFLOW_FROM_WORKFLOW != '' }}
-    - name: Checkout with token
-      if: steps.has-token.outputs.has-token == 'true'
-      uses: actions/checkout@v4
+    - uses: actions/checkout@v4
       with:
-        persist-credentials: true
-        ref: ${{ github.head_ref }}
-        fetch-depth: 3
-        token: ${{ secrets.RUN_WORKFLOW_FROM_WORKFLOW }}
-    - name: Checkout without token
-      if: steps.has-token.outputs.has-token != 'true'
+        ref: ${{ github.base_ref }}
+    - id: get-token
+      uses: ./.github/actions/get-token
+      with:
+        token-secret: ${{ secrets.RUN_WORKFLOW_FROM_WORKFLOW }}
```

Comment:
Weird space?

Reply:
I don't see any weird spacing? In JSON, this would be something like:
```json
{
  …,
  "steps": [
    { … }
    {
      "id": "get-token",
      "uses": "./.github/actions/get-token",
      "with": {
        "token-secret": "${{ secrets.RUN_WORKFLOW_FROM_WORKFLOW }}"
      }
    },
    { "name": "Checkout",
…
```